fabric-rendering-v1/src/client/java/net/fabricmc/fabric/api/client/rendering/v1/FabricLayeredDrawer.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.rendering.v1;
+
+import java.util.function.BooleanSupplier;
+import java.util.function.Function;
+
+import net.minecraft.client.gui.LayeredDrawer;
+import net.minecraft.util.Identifier;
+
+/**
+ * A layered drawer that has an identifier attached to each layer and methods to add layers in specific positions.
+ */
+public interface FabricLayeredDrawer {
```

Comment:
I think this should be renamed, its not a `LayeredDrawer` at all, its a facade over the vanilla one to hide the complexity. `LayeredDrawerContext`, `LayeredDrawerManager`, `LayeredDrawerModifications` Im not really too sure of a good name, I wonder if anyone else has a good suggestion.

Reply:
`LayeredDrawerWrapper` or `FabricLayeredDrawerWrapper`?

Reply:
`LayeredDrawerWrapper` is fine, we tend to only use `Fabric` when its an extended version of a vanilla class. In this case its not the vanilla class at all.